sources/api/migration/migrator/src/main.rs

```diff
@@ -23,45 +23,89 @@
 #[macro_use]
 extern crate log;
 
+use args::Args;
+use error::Result;
 use nix::{dir::Dir, fcntl::OFlag, sys::stat::Mode, unistd::fsync};
 use rand::{distributions::Alphanumeric, thread_rng, Rng};
 use semver::Version;
 use simplelog::{Config as LogConfig, TermLogger, TerminalMode};
 use snafu::{ensure, OptionExt, ResultExt};
 use std::collections::HashSet;
 use std::env;
-use std::fs::{self, Permissions};
+use std::fs::{self, File, OpenOptions, Permissions};
 use std::os::unix::fs::{symlink, PermissionsExt};
 use std::os::unix::io::AsRawFd;
 use std::path::{Path, PathBuf};
 use std::process::{self, Command};
+use tempfile::TempDir;
+use tough::ExpirationEnforcement;
+use update_metadata::{load_manifest, Direction, MIGRATION_FILENAME_RE, REPOSITORY_LIMITS};
+
+#[macro_use]
+extern crate lazy_static;
 
-use update_metadata::MIGRATION_FILENAME_RE;
 mod args;
-mod direction;
 mod error;
 
-use args::Args;
-use direction::Direction;
-use error::Result;
+lazy_static! {
+    /// This is the last version of Bottlerocket that supports *only* unsigned migrations.
+    static ref LAST_UNSIGNED_MIGRATIONS_VERSION: Version = Version::new(0, 3, 4);
```

Comment:
Unfortunately the compiler yelled at me because there is a Vec or a String inside (because they support suffixes like "alpha"). So allocation occurs. If it's too distasteful, I could just inline the ::new or return the Version from a function.

Reply:
Oh, right, this is `semver::Version`. See also https://github.com/steveklabnik/semver/issues/200#issuecomment-646316701